bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/ghost/GhostEditor.scala

```diff
@@ -56,87 +59,99 @@ final class GhostEditor extends ComponentEditor[ISPObsComponent, Ghost] {
       anchor = Anchor.NorthWest
-      gridx = 2
-      gridy = 3
-      insets = new Insets(20, 0, 0, 20)
+      gridx = 1
+      gridwidth = 2
+      gridy = 2
+      insets = new Insets(10, 0, 0, 20)
     }
 
     layout(new Label) = new Constraints() {
       anchor = Anchor.North
-      gridy = 4
+      gridy = 3
       weighty = 1.0
     }
 
-    listenTo(standardResolution, highResolution, asterismComboBox)
+    listenTo(asterismComboBox.selection)
     reactions += {
-      case ButtonClicked(`standardResolution`) =>
-        showStandardResolutionComponents(true)
-      case ButtonClicked(`highResolution`) =>
-        showStandardResolutionComponents(false)
+      case SelectionChanged(`asterismComboBox`) =>
+        val converter = asterismMap(asterismComboBox.selection.item)
+        convertAsterism(converter)
     }
 
-    def initialize(): Unit = {
-      val isHighResolution = Option(getContextTargetEnv).exists(_.getAsterism.isInstanceOf[GhostAsterism.HighResolution])
-      if (isHighResolution)
-        highResolution.doClick()
-      else
-        standardResolution.doClick()
+    /** Convert the asterism to the new type, and set the new target environment. */
+    def convertAsterism(converter: GhostAsterismConverter): Unit = Swing.onEDT {
+      val o = getContextObservation
+      val oc = getContextTargetObsComp
+
+      val newToc = for {
+        toc <- o.findTargetObsComp
+        env <- converter.convert(toc.getTargetEnvironment)
+      } yield {
+        toc.setTargetEnvironment(env)
+        toc
+      }
+
+      newToc match {
+        case Some(toc) =>
+          oc.setDataObject(toc)
+        case None =>
+          Dialog.showMessage(ui, "Could not convert the asterism type.",
+              "Asterism Conversion Error", Dialog.Message.Error)
+      }
     }
 
-    def showStandardResolutionComponents(viewable: Boolean): Unit = {
-      asterismLabel.visible = viewable
-      asterismComboBox.visible = viewable
+    def initialize(): Unit = Swing.onEDT {
+      // Set the combo box to the appropriate asterism type.
+      deafTo(asterismComboBox.selection)
+      ui.asterismComboBox.enabled = true
+      val selectionIdx = getContextObservation.findTargetObsComp.flatMap { env =>
+        env.getAsterism match {
+          case a: StandardResolution =>
+            a.targets match {
+              case SingleTarget(_)     => Some(0)
+              case DualTarget(_, _)    => Some(1)
+              case TargetPlusSky(_, _) => Some(2)
+              case SkyPlusTarget(_, _) => Some(3)
+            }
+          case HighResolution(_, _, _) => Some(4)
+          case _                       =>
+            Dialog.showMessage(ui, "Illegal asterism type for GHOST observation.",
+              "Asterism Error", Dialog.Message.Error)
+            ui.asterismComboBox.enabled = false
+            None
+        }
+      }
+      asterismComboBox.selection.index = selectionIdx.getOrElse(0)
```

Comment:
What is in the combo box model?  I wonder if you could handle it the same way you did for the position angle constraint combo in `PositionAnglePanel` where the model is the list of enum values.  Then when it comes time to change the selection it you can set the `selection.item` to an actual value instead of having to use the index:

`ui.positionAngleConstraintComboBox.selection.item = instrument.getPosAngleConstraint`

Reply:
There we go. Now `AsterismType`, which you can get from `Asterism`, is much more useful: it bundles together the `tag` for XML generation, the `name` of the asterism type, an empty `Asterism` creator for each type, and the `AsterismConverter` that is used to transform another `Asterism` into that type.

I like this much better, as `AsterismType` wasn't very useful before and only would have been used for P2 checks, really, and some very basic XML generation stuff.